### DISCUSSION TITLE
jdk threadpool plugin does not intercept objects implementing the RunnableFuture interface

### DISCUSSION THREAD

**Discussion by sun-rui** [apache/skywalking#9401] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

For a traced Java application, skywalking can't trace the span correctly for some cross-thread operations via ThreadPoolExecutor.

### What you expected to happen

skywalking should successfully trace the span of the cross-thread operations via ThreadPoolExecutor.

### How to reproduce

for example logic like the follows:
        ExecutorService executorService = Executors.newCachedThreadPool();

        Callable callable = call() -> { return something.execute(); };
        FutureTask futureTask = new FutureTask(callable);
        executorService.execute(futureTask);

Skylaking can't successfully trace the span.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by sun-rui** [comment]
The reason is the following:
```
public class ThreadPoolExecuteMethodInterceptor extends AbstractThreadingPoolInterceptor {

    @Override
    public Object wrap(Object param) {
        ...

        if (param instanceof RunnableFuture) {
            return null;
        }
```
I don't know why objects implementing the RunnableFuture interface are excluded for interception. After I comment out the if statement , skywalking can trace the span.

**Comment by wu-sheng** [comment]
First of all, this is not a bug.
This IF means, the RunnableFuture is not considered. How to deal with RunnableFuture would depend on the Java doc guideline.

**Reply by wu-sheng** [reply]
The open source software relies on the community to contribute. If you want to add this into supported, you need to follow plugin dev doc, and add relative cases to show how this works, and verify by plugin test tool.

We will review from there.

**Comment by sun-rui** [comment]
"How to deal with RunnableFuture would depend on the Java doc guideline." , sorry, I don't understand it.
Actually, the interceptor just cares the parameter object is a Runnable, but need not care about what subclass or sub-interface   of Runnable the object is of.  The point here is that the motivation of this special treatment on RunnableFuture which extends Runnable is unknown. This treatment put an unnecessary restriction on the usage scenarios of the cross-thread tracing capability of Skywalking.

**Reply by wu-sheng** [reply]
I think you still should understand how open-source project works. As a plugin, we as maintainers don't care about that detail thing. It was contributed by an individual and works for that individual's scenario. That is all.

There is no `need to care` or `don't need to care`. 

If you want to propose some changes, you need to prepare a pull request with test scenarios. Once we could see the changes and tests there, we could review them.
